devicetypes/heltun/heltun-hls01-switch.src/heltun-hls01-switch.groovy

```diff
@@ -0,0 +1,302 @@
+ name: "Selected Power Threshold in watts", paramNum: 26, size: 2, default: 0, type: "number", min: 0 , max: 3500, unit: "W"],
+
+[title: "High Load Timeout Protection: Time Threshold", description: "If High Load Timeout Protection is activated: Power Threshold is enabled, use this parameter to set the threshold value in minutes. If the load is connected longer than this value, the device will automatically switch Off the socket. Use the value 0 if there is a need to disable this function.",
+ name: "Selected Time Threshold in minutes", paramNum: 27, size: 2, default: 0, type: "number", min: 0 , max: 1440, unit: "min"],
+ 
+[title: "External Input: Hold Control Mode", description: "This Parameter defines how the relay should react while holding the button connected to the external input. The options are: Hold is disabled, Operate like click, Momentary Switch: When the button is held, the relay output state is ON, as soon as the button is released the relay output state changes to OFF, Reversed Momentary: When the button is held, the relay output state is OFF, as soon as the button is released the relay output state changes to ON.",
+ name: "Selected Hold Control Mode", options: [
+ 			0: "Hold is disabled",
+    		1: "Operate like click",
+			2: "Momentary Switch",
+            3: "Reversed Momentary"
+    ], paramNum: 41, size: 1, default: "2", type: "enum"],
+
+[title: "Hold Mode Duration for External Input S1", description: "This parameter specifies the time the device needs to recognize a hold mode when the button connected to an external input is held (key closed). This parameter is available on firmware V1.3 or higher",
+ name: "Selected Duration in milliseconds", paramNum: 46, size: 2, default: 500, type: "number", min: 200 , max: 5000, unit: "ms"],
+
+[title: "External Input: Click Control Mode", description: "This Parameter defines how the relay should react when clicking the button connected to the external input. The options are: Click is disabled, Toggle switch: relay inverts state (ON to OFF, OFF to ON), Only On: Relay switches to ON state only, Only Off: Relay switches to OFF state only, Timer: On > Off: Relay output switches to ON state (contacts are closed) then after a specified time switches back to OFF state (contacts are open). The time is specified in 'Relay Timer Mode Duration' below, Timer: Off > On: Relay output switches to OFF state (contacts are open) then after a specified time switches back to On state (contacts are closed). The time is specified in 'Relay Timer Mode Duration' below ",
+ name: "Selected Hold Control Mode", options: [
+ 			0: "Click is disabled",
+    		1: "Toggle Switch",
+			2: "Only On",
+            3: "Only Off",
+            4: "Timer: On > Off",
+            5: "Timer: Off > On"
+    ], paramNum: 51, size: 1, default: "1", type: "enum"],
```

Comment:
Please reformat this section.

Reply:
I have a question I don't know to whom I can ask, could you please answer the following question or let me know to whom can I ask?

I'm trying to create a device handler for a device that has 5 buttons and the supportedButtonValues are pushed, held, and released.

I am not able to set the supported button values correctly, even when I send "sendEvent(name: "supportedButtonValues", value: ["pushed", "held", "up"].encodeAsJSON(), displayed: false)" command to the child device.

I saw that there is a way by adding (vid: "generic-4-button") to the definition however there is no "generic-5-button" and it seems like the supportedButtonValues are predefined (I cannot change them)

Could you please help?

Reply:
@k-morawski @greens @KKlimczukS @moorage 

I have a question I don't know to whom I can ask, could you please answer the following question or let me know to whom can I ask?

I'm trying to create a device handler for a device that has 5 buttons and the supportedButtonValues are pushed, held, and released.

I am not able to set the supported button values correctly, even when I send "sendEvent(name: "supportedButtonValues", value: ["pushed", "held", "up"].encodeAsJSON(), displayed: false)" command to the child device.

I saw that there is a way by adding (vid: "generic-4-button") to the definition however there is no "generic-5-button" and it seems like the supportedButtonValues are predefined (I cannot change them)

Could you please help?

Reply:
@Sarkis008 As far as I know there is no generic-{X}-button for supporting X > 4 child devices (in one device detail page). But You could try adding more child devices and have them separately on the dashboard.
I'd like to see how do You create those child devices in DTH, so please open a separate PR with that code so we could discuss it there.